packages/common/src/services/audius-backend/AudiusBackend.ts

```diff
@@ -1618,74 +1634,20 @@ export const audiusBackend = ({
       }
 
       newMetadata = schemas.newUserMetadata(newMetadata, true)
-      const { blockHash, blockNumber, userId } =
-        await audiusLibs.User.updateCreator(
-          newMetadata.user_id,
-          newMetadata,
-          writeMetadataThroughChainEnabled
-        )
-      return { blockHash, blockNumber, userId }
-    } catch (err) {
-      console.error(getErrorMessage(err))
-      return false
-    }
-  }
-
-  // TODO (theo): don't call uploadImage if storage v2 signup is enabled
-  async function updateUser(metadata: User, id: ID) {
-    let newMetadata = { ...metadata }
-    const writeMetadataThroughChainEnabled =
-      (await getFeatureEnabled(FeatureFlags.WRITE_METADATA_THROUGH_CHAIN)) ??
-      false
-    try {
-      if (newMetadata.updatedProfilePicture) {
-        const resp = await audiusLibs.File.uploadImage(
-          newMetadata.updatedProfilePicture.file,
-          undefined,
-          null,
-          writeMetadataThroughChainEnabled
-        )
-        newMetadata.profile_picture_sizes = resp.dirCID
-      }
-
-      if (newMetadata.updatedCoverPhoto) {
-        const resp = await audiusLibs.File.uploadImage(
-          newMetadata.updatedCoverPhoto.file,
-          false,
-          null,
-          writeMetadataThroughChainEnabled
-        )
-        newMetadata.cover_photo_sizes = resp.dirCID
-      }
-      if (
-        typeof newMetadata.twitter_handle === 'string' ||
-        typeof newMetadata.instagram_handle === 'string' ||
-        typeof newMetadata.tiktok_handle === 'string' ||
-        typeof newMetadata.website === 'string' ||
-        typeof newMetadata.donation === 'string'
-      ) {
-        await fetch(`${identityServiceUrl}/social_handles`, {
-          method: 'POST',
-          headers: {
-            'Content-Type': 'application/json'
-          },
-          body: JSON.stringify({
-            handle: newMetadata.handle,
-            twitterHandle: newMetadata.twitter_handle,
-            instagramHandle: newMetadata.instagram_handle,
-            website: newMetadata.website,
-            donation: newMetadata.donation
-          })
-        })
+      if (storageV2UploadEnabled) {
+        const userId = newMetadata.user_id
+        const { blockHash, blockNumber } =
+          await audiusLibs.User.updateMetadataV2({ newMetadata, userId })
+        return { blockHash, blockNumber, userId }
+      } else {
+        const { blockHash, blockNumber, userId } =
+          await audiusLibs.User.updateCreator(
+            newMetadata.user_id,
+            newMetadata,
+            writeMetadataThroughChainEnabled
+          )
+        return { blockHash, blockNumber, userId }
       }
-
-      newMetadata = schemas.newUserMetadata(newMetadata, true)
-
-      const { blockHash, blockNumber } = await audiusLibs.User.updateUser(
```

Comment:
this was removed from libs in [here](https://github.com/AudiusProject/audius-protocol/pull/5012/files#diff-6cf87bbc23e9cdb74c8f51008bafae99d2d6f17a1193a4999f17ee0ed1842653L637), so this whole function would always error. I only saw 2 instances of "function not found" in Sentry, though, because this path is only called when a user doesn't have a replica set, which should never happen because sanity checks on login should ensure they have one